ledger/src/blockstore.rs

```diff
@@ -524,16 +524,46 @@ macro_rules! rocksdb_metric_header {
     ($metric_name:literal, $cf_name:literal, $column_options:expr) => {
         match $column_options.shred_storage_type {
             ShredStorageType::RocksLevel =>
-                rocksdb_metric_header!(@all_fields $metric_name, $cf_name, "rocks_level"),
+                rocksdb_metric_header!(@compression_type $metric_name, $cf_name, $column_options, "rocks_level"),
             ShredStorageType::RocksFifo(_) =>
-                rocksdb_metric_header!(@all_fields $metric_name, $cf_name, "rocks_fifo"),
+                rocksdb_metric_header!(@compression_type $metric_name, $cf_name, $column_options, "rocks_fifo"),
         }
     };
 
-    (@all_fields $metric_name:literal, $cf_name:literal, $storage_type:literal) => {
+    (@compression_type $metric_name:literal, $cf_name:literal, $column_options:expr, $storage_type:literal) => {
+        match $column_options.compression_type {
```

Comment:
Not sure if there is a way around it, but these match statement are a bit repetitive. Maybe we keep an eye on it, try to figure out something better if they start to get out of hand with more "group by"

Reply:
Not sure why my previous replay disappears xD.  Let me comment again.

I think the ideal way is to improve the datapoint macro to take an optional number of tags, something like the following:

```
    ($name:expr, $(tags:tt)+, $($fields:tt)+)
```

But I'm not sure how we can distinguish `tags` and `fields` given that `tags` are optional and could have arbitrary number of them.